rfcs/text/0003_handler_interface.md

```diff
@@ -297,9 +308,14 @@ class MyPlugin {
   necessarily associate similar patterns elsewhere as the same set of problems.
 - "Chicken and egg" questions will arise around where context providers should be
   registered. For example, does the `http` service invoke its
-  registerCapabilities for `elasticsearch`, or does the `elasticsearch` service
-  invoke `http.registerCapabilities`, or does core itself register the
+  registerRequestContext for `elasticsearch`, or does the `elasticsearch` service
+  invoke `http.registerRequestContext`, or does core itself register the
   provider so neither service depends directly on the other.
+- The existence of plugins that a given plugin does not depend on may leak
```

Comment:
Since this RFC is about a broad concept rather than a specific feature, I'm fine proceeding with this RFC as-is, but this detail is an important problem to address for implementations. Realistically, I think this means that core must provide plugin developers a mechanism to build a context since by nature a plugin won't even know about plugins that it does not depend on, so they can't properly construct the context themselves.

Reply:
Agreed. We may actually need to expose a service to plugins that allows them to offer context registration and construction. This service would handle the dependency management aspect of constructing context.

This needs to be built before any plugins begin offering context in their handlers.